src/transport_performance/analyse_network.py

```diff
@@ -0,0 +1,441 @@
+                    "a valid GTFS."
+                )
+
+        return travel_time_matrix_computer.compute_travel_times()
+
+    def _gdf_batch_origins(
+        self,
+        gdf: gpd.GeoDataFrame,
+        destination_col: str,
+        distance: float,
+        num_origins: int,
+        unit: Unit = Unit.KILOMETERS,
+    ) -> (list, list):
+        """Split geopandas.DataFrame into batches of a single origin.
+
+        Parameters
+        ----------
+        gdf : gpd.GeoDataFrame
+            Geodataframe containing location ids to cross match.
+        destination_col : str
+            Column indicating what centroids should be considered as
+            destinations.
+        distance : float
+            Distance to filter destinations. Points further away from
+            origin are removed from output.
+        num_origins : int
+            Number of origins to consider. Note that more origins will greatly
+            increase the cartesian product of origins and destinations, which
+            may slow down processing or exceed available memory. To use all
+            possible origins, use `len(gdf)`.
+        unit : Unit
+            Unit to calculate distance. Default is km.
+
+        Yields
+        ------
+        sel_origins : list
+            Id numbers of the origins in a given iteration.
+        sel_dest : list
+            Id numbers of the destinations in a given iteration.
+
+        Raises
+        ------
+        ValueError
+            If the number of origins provided is not between 1 and the total
+            number of origins in the geodataframe.
+
+        Notes
+        -----
+        This function will filter out pairs of origins and destinations that
+        are over the distance threshold, and it will return a list with
+        unique origins and filtered destinations. However, note that this will
+        only work as intended if using a single origin. If using several
+        origins per batch, the r5py `TravelTimeMatrixComputer` object will
+        still create all possible combinations of origins and destinations
+        provided, which may include pairs that are beyond the threshold.
+
+        """
+        # defences
+        d._type_defence(gdf, "gdf", gpd.GeoDataFrame)
+        d._check_column_in_df(gdf, destination_col)
+        d._type_defence(distance, "distance", float)
+        d._type_defence(num_origins, "num_origins", int)
+        d._type_defence(unit, "unit", Unit)
+        if num_origins < 1 or num_origins > len(gdf):
+            raise ValueError(
+                f"`num_origins` should be between 1 and {len(gdf)}, "
+                f"got {num_origins}"
+            )
+
+        # get sources and destinations
+        # define destinations when `destination_col` is true
+        orig_gdf = gdf.copy()
+        geometry_col = gdf.geometry.name
+        # TODO: add option to include all rows as destinations?
+        dest_gdf = (
```

Comment:
Added check for `bool dtype`. Added tests.

Reply:
Added commit `5a7b448` to move these checks to the start of `od_matrix()` so that the defence checks are done for both batched and non-batched modes. Unit tests updated to reflect this change too. Tested locally, and expected errors are being raised as needed. Happy to close this conversation. Thanks again for your help!